module/os/linux/zfs/zpl_xattr.c

```diff
@@ -153,26 +153,28 @@ static int
 zpl_xattr_readdir(struct inode *dxip, xattr_filldir_t *xf)
 {
 	zap_cursor_t zc;
-	zap_attribute_t	zap;
+	zap_attribute_t	*zap = zap_attribute_alloc();
```

Comment:
Again to the question of extended attribute names length limit.

Reply:
xattr name length is limit by kernel to 255

Reply:
OK if so, I just haven't found the check on my quick search for it.

Reply:
https://elixir.bootlin.com/linux/v6.11/source/include/uapi/linux/limits.h#L15